### PR TITLE
refactor: Easily change complex logic

### PR DESCRIPTION
**Pull request by birariro**
- Removed complex branching statements from getIdentifierColumns()
  - No need to check that "joins" size is 1

- Changed the logic of "handleSelect()" to be more readable

### CODE REVIEW

File: querydsl-sql/src/main/java/com/querydsl/sql/SQLSerializer.java [hunk 1]
```diff
@@ -115,41 +115,28 @@ public List<Path<?>> getConstantPaths() {
      */
     @SuppressWarnings("unchecked")
     protected List<Expression<?>> getIdentifierColumns(List<JoinExpression> joins, boolean alias) {
-        if (joins.size() == 1) {
-            JoinExpression join = joins.get(0);
-            if (join.getTarget() instanceof RelationalPath) {
-                return ((RelationalPath) join.getTarget()).getColumns();
```
Comment (jwgmeligmeyling) [inline querydsl-sql/src/main/java/com/querydsl/sql/SQLSerializer.java:121]: This optimisation is gone.

File: querydsl-sql/src/main/java/com/querydsl/sql/SQLSerializer.java [hunk 2]
```diff
@@ -115,41 +115,28 @@ public List<Path<?>> getConstantPaths() {
...
 
-                } else {
-                    // not able to provide a distinct list of columns
-                    return Collections.emptyList();
-                }
-            }
-            return rv;
-        }
+			RelationalPath<?> path = (RelationalPath<?>)join.getTarget();
+			List<Expression<?>> columns =
```
Comment (jwgmeligmeyling) [inline querydsl-sql/src/main/java/com/querydsl/sql/SQLSerializer.java:127]: Ternary operators are arguably less readable

File: querydsl-sql/src/main/java/com/querydsl/sql/SQLSerializer.java [hunk 3]
```diff
@@ -162,22 +149,22 @@ public void handle(String template, Object... args) {
...
-                        expr = ExpressionUtils.as(expr, "col__" + name + replacements.size());
-                    }
-                }
-                replacements.add(expr);
-            }
-            return handle(sep, replacements);
-        } else {
-            return handle(sep, expressions);
-        }
+        if (!inSubquery) {
```
Comment (jwgmeligmeyling) [inline querydsl-sql/src/main/java/com/querydsl/sql/SQLSerializer.java:152]: Negative conditionals are not considered more readable.

**Comment by jwgmeligmeyling** [comment]
The code worsened in three places. Why fix whats not broken?

File: querydsl-sql/src/main/java/com/querydsl/sql/SQLSerializer.java (hunk 1 above)
Comment (birariro) [reply querydsl-sql/src/main/java/com/querydsl/sql/SQLSerializer.java:121]: I didn't realize it was an optimization.

File: querydsl-sql/src/main/java/com/querydsl/sql/SQLSerializer.java (hunk 3 above)
Comment (birariro) [reply querydsl-sql/src/main/java/com/querydsl/sql/SQLSerializer.java:152]: Negative conditional statements are not easier to read.
They just make the code easier to read by reducing its depth.

You might want to check out
https://go.dev/doc/effective_go#if
https://stackoverflow.com/questions/2928556/programming-style-should-you-return-early-if-a-guard-condition-is-not-satisfied

**Pull request closed without merging**